packages/workers-shared/asset-worker/src/handler.ts

```diff
@@ -692,6 +700,14 @@ export const decodePath = (pathname: string) => {
 const encodePath = (pathname: string) => {
 	return pathname
 		.split("/")
-		.map((x) => encodeURIComponent(x))
+		.map((x) => {
+			let encoded;
+			try {
+				encoded = encodeURIComponent(x);
```

Comment:
Can encodeURIComponent actually throw?

Reply:
If given something like a lone surrogate, yes:

`encodeURIComponent("\uD800")`

How likely/possible this is to happen in a URL? Probably not, but I felt like it couldn't hurt to wrap both.